base/linalg/dense.jl

```diff
@@ -29,6 +33,41 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    if m < div(ISONE_CUTOFF * m * m, sizeof(x))
```

Comment:
Oh, sorry, when reading this part I was taking `m` as the length of the matrix :-/
Agreed, `__CUP_CACHE_SIZE__` would be useful, I for one had to roll my own version for some array algorithms. In the meantime, `sizeof(x) < ISONE_CUTOFF` will do.